TypeCobol/Compiler/Diagnostics/DiagnosticMessage.cs

```diff
@@ -52,12 +52,16 @@ internal DiagnosticMessage(Category category, int code, Severity severity, strin
 
         // Load messages table DiagnosticMessages.csv in memory at startup
 
-        public static IList<DiagnosticMessage> GetFromCode { get; private set; }
+        private static readonly Dictionary<MessageCode, DiagnosticMessage> _predefinedMessages;
+        public static DiagnosticMessage GetFromCode(MessageCode code)
+        {
+            System.Diagnostics.Debug.Assert(_predefinedMessages.ContainsKey(code));
+            return _predefinedMessages[code];
+        }
 
         static DiagnosticMessage()
```

Comment:
I think is time to make this static constructor thread safe, because I think that the first thread that will create a diagnostic will execute this code, but if will filling _predefinedMessages another thread creates a diagnostic there will be a concurrent access here, for instance LSR and Analyzers can be in differents threads, but's ok if it is same thread that run analyzers in the QualityCheck step. No other thread should create a diagnostics outside the compiler thread steps.

Reply:
Well, we're in luck then because static constructors are thread-safe in .NET. I could not find the doc reference but this is discussed [here](https://github.com/dotnet/docs/issues/10243). Basically first thread that needs to use the type will run the static constructor and other threads are on hold until the type is properly initialized.

Reply:
Ok so that's good.